src/background.js

```diff
@@ -28,6 +34,46 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc)
+    if (samePageCandidates.length <= 0) {
+        try {
+            const pageData = await fetchPageData(bookmarkInfo.url, {
+                                                    includeFreezeDry: true,
+                                                    includePageContent: true,
+                                                    includeFavIcon: true,
+                                                })
+            const favIconBlob = await dataURLToBlob(pageData.favIconURI)
+            const freezeDryBlob = new Blob([pageData.freezeDryHTML], {type: 'text/html;charset=UTF-8'})
+            const pageDoc = {
+                url: bookmarkInfo.url,
+                title: bookmarkInfo.title,
+                content: pageData.content,
+                _id: generatePageDocId(bookmarkInfo.dateAdded, id),
+                _attachments: [{
+                        content_type: freezeDryBlob.type,
+                        data: freezeDryBlob
+                    }, {
+                        content_type: favIconBlob.type,
+                        data: favIconBlob
+                    }
+                ]
+            }
+        } catch (err) {
+            console.log("Error occurred while fetching page data: " + err.toString())
+            const pageDoc = {
+                url: bookmarkInfo.url,
+                title: bookmarkInfo.title,
+                _id: generatePageDocId(bookmarkInfo.dateAdded, id),
+            }
+        } finally {
+            const bookmarkDoc = await transformToBookmarkDoc(pageDoc, bookmarkInfo)
+            db.bulkDocs([bookmarkDoc, pageDoc])
```

Comment:
`pageDoc` is still not in scope here. `try`, `catch`, `finally` blocks all have their own scope, and don't overlap. Best thing just declare it before the `try` block (in the inherited parent scope) as `let pageDoc = { url: ..., ... }`, then add content and attachments to the object in the `try` block. `catch` block shouldn't need anything - if it fails, we can't and don't need to do anything.

Also you should put this entire `addListener` callback in its own function somewhere and then you could test it out in dev by importing it from the UI and making a simple button to call it.